xbmc/filesystem/VideoDatabaseDirectory/DirectoryNode.cpp

```diff
@@ -50,14 +50,14 @@ using namespace XFILE::VIDEODATABASEDIRECTORY;
 //  Constructor is protected use ParseURL()
 CDirectoryNode::CDirectoryNode(NODE_TYPE Type, const std::string& strName, CDirectoryNode* pParent)
 {
-  m_Type=Type;
-  m_strName=strName;
-  m_pParent=pParent;
+  m_Type = Type;
+  m_strName = strName;
+  m_pParent = pParent;
 }
 
 CDirectoryNode::~CDirectoryNode()
 {
-  delete m_pParent;
+  delete m_pParent, m_pParent = nullptr;
```

Comment:
If that hides threading issues, e.g. another thread is accessing the member while the dtor is running, we needed proper locking here.

Other than that - we don't care about it after the dtor is finished.